crates/iceberg/src/spec/transform.rs

```diff
@@ -398,6 +679,80 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_none_projection() -> Result<()> {
+        let name = "projected_name".to_string();
+        let preds = TestPredicates::new();
+
+        let transform = Transform::Void;
+        let result_unary = transform.project(name.clone(), &preds.unary)?;
+        assert!(result_unary.is_none());
+
+        let transform = Transform::Year;
+        let result_binary = transform.project(name.clone(), &preds.binary)?;
+        assert!(result_binary.is_none());
+
+        let transform = Transform::Month;
+        let result_set = transform.project(name.clone(), &preds.set)?;
+        assert!(result_set.is_none());
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_truncate_project() -> Result<()> {
+        let name = "projected_name".to_string();
+        let preds = TestPredicates::new();
+
+        let transform = Transform::Truncate(10);
+
+        let result_unary = transform.project(name.clone(), &preds.unary)?.unwrap();
+        let result_binary = transform.project(name.clone(), &preds.binary)?.unwrap();
+        let result_set = transform.project(name.clone(), &preds.set)?.unwrap();
+
+        assert_eq!(format!("{}", result_unary), "projected_name IS NULL");
+        assert_eq!(format!("{}", result_binary), "projected_name = 0");
+        assert_eq!(format!("{}", result_set), "projected_name IN (0)");
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_identity_project() -> Result<()> {
+        let name = "projected_name".to_string();
+        let preds = TestPredicates::new();
+
+        let transform = Transform::Identity;
+
+        let result_unary = transform.project(name.clone(), &preds.unary)?.unwrap();
+        let result_binary = transform.project(name.clone(), &preds.binary)?.unwrap();
+        let result_set = transform.project(name.clone(), &preds.set)?.unwrap();
+
+        assert_eq!(format!("{}", result_unary), "projected_name IS NULL");
+        assert_eq!(format!("{}", result_binary), "projected_name = 5");
+        assert_eq!(format!("{}", result_set), "projected_name IN (5, 6)");
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_bucket_project() -> Result<()> {
+        let name = "projected_name".to_string();
+        let preds = TestPredicates::new();
+
+        let transform = Transform::Bucket(8);
+
+        let result_unary = transform.project(name.clone(), &preds.unary)?.unwrap();
+        let result_binary = transform.project(name.clone(), &preds.binary)?.unwrap();
+        let result_set = transform.project(name.clone(), &preds.set)?.unwrap();
+
+        assert_eq!(format!("{}", result_unary), "projected_name IS NULL");
+        assert_eq!(format!("{}", result_binary), "projected_name = 7");
+        assert_eq!(format!("{}", result_set), "projected_name IN (1, 7)");
+
+        Ok(())
+    }
+
```

Comment:
I think the most important tests are missing around the dates projection: https://github.com/apache/iceberg/blob/main/api/src/test/java/org/apache/iceberg/transforms/TestDatesProjection.java